// @medusajs/medusa/dist/api-v2/admin/payments/query-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.retrieveTransformQueryConfig = exports.listTransformQueryConfig = exports.allowedAdminPaymentRelations = exports.defaultAdminPaymentRelations = exports.defaultAdminPaymentFields = void 0;
exports.defaultAdminPaymentFields = [
    "id",
    "currency_code",
    "amount",
    "captured_at",
    "payment_collection_id",
    "payment_session_id",
    "captures.id",
    "captures.amount",
    "refunds.id",
    "refunds.amount",
];
exports.defaultAdminPaymentRelations = ["captures", "refunds"];
exports.allowedAdminPaymentRelations = ["captures", "refunds"];
exports.listTransformQueryConfig = {
    defaultFields: exports.defaultAdminPaymentFields,
    defaultRelations: exports.defaultAdminPaymentRelations,
    allowedRelations: exports.allowedAdminPaymentRelations,
    isList: true,
};
exports.retrieveTransformQueryConfig = {
    defaultFields: exports.defaultAdminPaymentFields,
    defaultRelations: exports.defaultAdminPaymentRelations,
    allowedRelations: exports.allowedAdminPaymentRelations,
    isList: false,
};
