// @medusajs/medusa/dist/api-v2/admin/payments/validators.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminPostPaymentsRefundsReq = exports.AdminPostPaymentsCapturesReq = exports.AdminGetPaymentsParams = exports.AdminGetPaymentsPaymentParams = void 0;
var class_transformer_1 = require("class-transformer");
var class_validator_1 = require("class-validator");
var common_1 = require("../../../types/common");
var utils_1 = require("../../../utils");
var AdminGetPaymentsPaymentParams = /** @class */ (function (_super) {
    __extends(AdminGetPaymentsPaymentParams, _super);
    function AdminGetPaymentsPaymentParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminGetPaymentsPaymentParams;
}(common_1.FindParams));
exports.AdminGetPaymentsPaymentParams = AdminGetPaymentsPaymentParams;
var AdminGetPaymentsParams = /** @class */ (function (_super) {
    __extends(AdminGetPaymentsParams, _super);
    function AdminGetPaymentsParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([String, [String]]),
        __metadata("design:type", Object)
    ], AdminGetPaymentsParams.prototype, "id", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)(),
        (0, class_transformer_1.Type)(function () { return common_1.DateComparisonOperator; }),
        __metadata("design:type", common_1.DateComparisonOperator
        /**
         * Date filters to apply on the customer users' `created_at` date.
         */
        )
    ], AdminGetPaymentsParams.prototype, "updated_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)(),
        (0, class_transformer_1.Type)(function () { return common_1.DateComparisonOperator; }),
        __metadata("design:type", common_1.DateComparisonOperator
        /**
         * Date filters to apply on the users' `deleted_at` date.
         */
        )
    ], AdminGetPaymentsParams.prototype, "created_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)(),
        (0, class_transformer_1.Type)(function () { return common_1.DateComparisonOperator; }),
        __metadata("design:type", common_1.DateComparisonOperator)
    ], AdminGetPaymentsParams.prototype, "deleted_at", void 0);
    return AdminGetPaymentsParams;
}((0, common_1.extendedFindParamsMixin)({
    limit: 20,
    offset: 0,
})));
exports.AdminGetPaymentsParams = AdminGetPaymentsParams;
var AdminPostPaymentsCapturesReq = /** @class */ (function () {
    function AdminPostPaymentsCapturesReq() {
    }
    __decorate([
        (0, class_validator_1.IsInt)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Number)
    ], AdminPostPaymentsCapturesReq.prototype, "amount", void 0);
    return AdminPostPaymentsCapturesReq;
}());
exports.AdminPostPaymentsCapturesReq = AdminPostPaymentsCapturesReq;
var AdminPostPaymentsRefundsReq = /** @class */ (function () {
    function AdminPostPaymentsRefundsReq() {
    }
    __decorate([
        (0, class_validator_1.IsInt)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Number)
    ], AdminPostPaymentsRefundsReq.prototype, "amount", void 0);
    return AdminPostPaymentsRefundsReq;
}());
exports.AdminPostPaymentsRefundsReq = AdminPostPaymentsRefundsReq;
