// @medusajs/medusa/dist/api-v2/admin/tax-regions/validators.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminPostTaxRegionsReq = exports.AdminGetTaxRegionsParams = exports.AdminGetTaxRegionsTaxRegionParams = void 0;
var class_transformer_1 = require("class-transformer");
var class_validator_1 = require("class-validator");
var common_1 = require("../../../types/common");
var operator_map_1 = require("../../../types/validators/operator-map");
var AdminGetTaxRegionsTaxRegionParams = /** @class */ (function (_super) {
    __extends(AdminGetTaxRegionsTaxRegionParams, _super);
    function AdminGetTaxRegionsTaxRegionParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminGetTaxRegionsTaxRegionParams;
}(common_1.FindParams));
exports.AdminGetTaxRegionsTaxRegionParams = AdminGetTaxRegionsTaxRegionParams;
var AdminGetTaxRegionsParams = /** @class */ (function (_super) {
    __extends(AdminGetTaxRegionsParams, _super);
    function AdminGetTaxRegionsParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, class_validator_1.IsString)({ each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminGetTaxRegionsParams.prototype, "id", void 0);
    __decorate([
        (0, class_validator_1.IsString)({ each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminGetTaxRegionsParams.prototype, "country_code", void 0);
    __decorate([
        (0, class_validator_1.IsString)({ each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminGetTaxRegionsParams.prototype, "province_code", void 0);
    __decorate([
        (0, class_validator_1.IsString)({ each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminGetTaxRegionsParams.prototype, "parent_id", void 0);
    __decorate([
        (0, class_validator_1.IsString)({ each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminGetTaxRegionsParams.prototype, "created_by", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)(),
        (0, class_transformer_1.Type)(function () { return operator_map_1.OperatorMapValidator; }),
        __metadata("design:type", Object)
    ], AdminGetTaxRegionsParams.prototype, "created_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)(),
        (0, class_transformer_1.Type)(function () { return operator_map_1.OperatorMapValidator; }),
        __metadata("design:type", Object)
    ], AdminGetTaxRegionsParams.prototype, "updated_at", void 0);
    __decorate([
        (0, class_validator_1.ValidateNested)(),
        (0, class_validator_1.IsOptional)(),
        (0, class_transformer_1.Type)(function () { return operator_map_1.OperatorMapValidator; }),
        __metadata("design:type", Object)
    ], AdminGetTaxRegionsParams.prototype, "deleted_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return AdminGetTaxRegionsParams; }),
        __metadata("design:type", Array)
    ], AdminGetTaxRegionsParams.prototype, "$and", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return AdminGetTaxRegionsParams; }),
        __metadata("design:type", Array)
    ], AdminGetTaxRegionsParams.prototype, "$or", void 0);
    return AdminGetTaxRegionsParams;
}((0, common_1.extendedFindParamsMixin)({
    limit: 50,
    offset: 0,
})));
exports.AdminGetTaxRegionsParams = AdminGetTaxRegionsParams;
var CreateDefaultTaxRate = /** @class */ (function () {
    function CreateDefaultTaxRate() {
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsNumber)(),
        __metadata("design:type", Object)
    ], CreateDefaultTaxRate.prototype, "rate", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", Object)
    ], CreateDefaultTaxRate.prototype, "code", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], CreateDefaultTaxRate.prototype, "name", void 0);
    __decorate([
        (0, class_validator_1.IsObject)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], CreateDefaultTaxRate.prototype, "metadata", void 0);
    return CreateDefaultTaxRate;
}());
var AdminPostTaxRegionsReq = /** @class */ (function () {
    function AdminPostTaxRegionsReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostTaxRegionsReq.prototype, "country_code", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostTaxRegionsReq.prototype, "province_code", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostTaxRegionsReq.prototype, "parent_id", void 0);
    __decorate([
        (0, class_validator_1.ValidateNested)(),
        (0, class_validator_1.IsOptional)(),
        (0, class_transformer_1.Type)(function () { return CreateDefaultTaxRate; }),
        __metadata("design:type", CreateDefaultTaxRate)
    ], AdminPostTaxRegionsReq.prototype, "default_tax_rate", void 0);
    __decorate([
        (0, class_validator_1.IsObject)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminPostTaxRegionsReq.prototype, "metadata", void 0);
    return AdminPostTaxRegionsReq;
}());
exports.AdminPostTaxRegionsReq = AdminPostTaxRegionsReq;
