// @medusajs/medusa/dist/api-v2/store/carts/validators.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorePostCartsCartTaxesReq = exports.StorePostCartsCartReq = exports.StoreDeleteCartsCartPromotionsReq = exports.StorePostCartsCartPromotionsReq = exports.StorePostCartReq = exports.Item = exports.StoreGetCartsCartParams = void 0;
var class_transformer_1 = require("class-transformer");
var class_validator_1 = require("class-validator");
var common_1 = require("../../../types/common");
var utils_1 = require("../../../utils");
var StoreGetCartsCartParams = /** @class */ (function (_super) {
    __extends(StoreGetCartsCartParams, _super);
    function StoreGetCartsCartParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return StoreGetCartsCartParams;
}(common_1.FindParams));
exports.StoreGetCartsCartParams = StoreGetCartsCartParams;
var Item = /** @class */ (function () {
    function Item() {
    }
    __decorate([
        (0, class_validator_1.IsNotEmpty)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], Item.prototype, "variant_id", void 0);
    __decorate([
        (0, class_validator_1.IsNotEmpty)(),
        (0, class_validator_1.IsInt)(),
        __metadata("design:type", Number)
    ], Item.prototype, "quantity", void 0);
    return Item;
}());
exports.Item = Item;
var StorePostCartReq = /** @class */ (function () {
    function StorePostCartReq() {
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], StorePostCartReq.prototype, "region_id", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([common_1.AddressPayload, String]),
        __metadata("design:type", Object)
    ], StorePostCartReq.prototype, "shipping_address", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([common_1.AddressPayload, String]),
        __metadata("design:type", Object)
    ], StorePostCartReq.prototype, "billing_address", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], StorePostCartReq.prototype, "email", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], StorePostCartReq.prototype, "currency_code", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsArray)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return Item; }),
        __metadata("design:type", Array)
    ], StorePostCartReq.prototype, "items", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], StorePostCartReq.prototype, "sales_channel_id", void 0);
    __decorate([
        (0, class_validator_1.IsObject)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], StorePostCartReq.prototype, "metadata", void 0);
    return StorePostCartReq;
}());
exports.StorePostCartReq = StorePostCartReq;
var StorePostCartsCartPromotionsReq = /** @class */ (function () {
    function StorePostCartsCartPromotionsReq() {
    }
    __decorate([
        (0, class_validator_1.IsArray)(),
        (0, class_transformer_1.Type)(function () { return String; }),
        __metadata("design:type", Array)
    ], StorePostCartsCartPromotionsReq.prototype, "promo_codes", void 0);
    return StorePostCartsCartPromotionsReq;
}());
exports.StorePostCartsCartPromotionsReq = StorePostCartsCartPromotionsReq;
var StoreDeleteCartsCartPromotionsReq = /** @class */ (function () {
    function StoreDeleteCartsCartPromotionsReq() {
    }
    __decorate([
        (0, class_validator_1.IsArray)(),
        (0, class_transformer_1.Type)(function () { return String; }),
        __metadata("design:type", Array)
    ], StoreDeleteCartsCartPromotionsReq.prototype, "promo_codes", void 0);
    return StoreDeleteCartsCartPromotionsReq;
}());
exports.StoreDeleteCartsCartPromotionsReq = StoreDeleteCartsCartPromotionsReq;
var StorePostCartsCartReq = /** @class */ (function () {
    function StorePostCartsCartReq() {
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], StorePostCartsCartReq.prototype, "region_id", void 0);
    __decorate([
        (0, class_validator_1.IsEmail)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], StorePostCartsCartReq.prototype, "email", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([common_1.AddressPayload, String]),
        __metadata("design:type", Object)
    ], StorePostCartsCartReq.prototype, "billing_address", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([common_1.AddressPayload, String]),
        __metadata("design:type", Object)
    ], StorePostCartsCartReq.prototype, "shipping_address", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], StorePostCartsCartReq.prototype, "sales_channel_id", void 0);
    __decorate([
        (0, class_validator_1.IsObject)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], StorePostCartsCartReq.prototype, "metadata", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsArray)(),
        (0, class_transformer_1.Type)(function () { return String; }),
        __metadata("design:type", Array)
    ], StorePostCartsCartReq.prototype, "promo_codes", void 0);
    return StorePostCartsCartReq;
}());
exports.StorePostCartsCartReq = StorePostCartsCartReq;
var StorePostCartsCartTaxesReq = /** @class */ (function () {
    function StorePostCartsCartTaxesReq() {
    }
    return StorePostCartsCartTaxesReq;
}());
exports.StorePostCartsCartTaxesReq = StorePostCartsCartTaxesReq;
