// @medusajs/medusa/dist/api-v2/admin/stores/validators.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminDeleteStoresStoreReq = exports.AdminPostStoresStoreReq = exports.AdminGetStoresParams = exports.AdminGetStoresStoreParams = void 0;
var class_transformer_1 = require("class-transformer");
var class_validator_1 = require("class-validator");
var common_1 = require("../../../types/common");
var AdminGetStoresStoreParams = /** @class */ (function (_super) {
    __extends(AdminGetStoresStoreParams, _super);
    function AdminGetStoresStoreParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminGetStoresStoreParams;
}(common_1.FindParams));
exports.AdminGetStoresStoreParams = AdminGetStoresStoreParams;
/**
 * Parameters used to filter and configure the pagination of the retrieved api keys.
 */
var AdminGetStoresParams = /** @class */ (function (_super) {
    __extends(AdminGetStoresParams, _super);
    function AdminGetStoresParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, class_validator_1.IsString)({ each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminGetStoresParams.prototype, "id", void 0);
    __decorate([
        (0, class_validator_1.IsString)({ each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminGetStoresParams.prototype, "name", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return AdminGetStoresParams; }),
        __metadata("design:type", Array)
    ], AdminGetStoresParams.prototype, "$and", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return AdminGetStoresParams; }),
        __metadata("design:type", Array)
    ], AdminGetStoresParams.prototype, "$or", void 0);
    return AdminGetStoresParams;
}((0, common_1.extendedFindParamsMixin)({
    limit: 50,
    offset: 0,
})));
exports.AdminGetStoresParams = AdminGetStoresParams;
var AdminPostStoresStoreReq = /** @class */ (function () {
    function AdminPostStoresStoreReq() {
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostStoresStoreReq.prototype, "name", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsArray)(),
        __metadata("design:type", Array)
    ], AdminPostStoresStoreReq.prototype, "supported_currency_codes", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostStoresStoreReq.prototype, "default_currency_code", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostStoresStoreReq.prototype, "default_sales_channel_id", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostStoresStoreReq.prototype, "default_region_id", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostStoresStoreReq.prototype, "default_location_id", void 0);
    __decorate([
        (0, class_validator_1.IsObject)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminPostStoresStoreReq.prototype, "metadata", void 0);
    return AdminPostStoresStoreReq;
}());
exports.AdminPostStoresStoreReq = AdminPostStoresStoreReq;
var AdminDeleteStoresStoreReq = /** @class */ (function () {
    function AdminDeleteStoresStoreReq() {
    }
    return AdminDeleteStoresStoreReq;
}());
exports.AdminDeleteStoresStoreReq = AdminDeleteStoresStoreReq;
