// @medusajs/medusa/dist/api-v2/admin/invites/query-config.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listTransformQueryConfig = exports.retrieveTransformQueryConfig = exports.defaultAdminInviteFields = exports.allowedAdminInviteRelations = exports.defaultAdminInviteRelations = void 0;
exports.defaultAdminInviteRelations = [];
exports.allowedAdminInviteRelations = [];
exports.defaultAdminInviteFields = [
    "id",
    "email",
    "accepted",
    "token",
    "expires_at",
    "metadata",
    "created_at",
    "updated_at",
    "deleted_at",
];
exports.retrieveTransformQueryConfig = {
    defaultFields: exports.defaultAdminInviteFields,
    defaultRelations: exports.defaultAdminInviteRelations,
    allowedRelations: exports.allowedAdminInviteRelations,
    isList: false,
};
exports.listTransformQueryConfig = __assign(__assign({}, exports.retrieveTransformQueryConfig), { isList: true });
