// @medusajs/medusa/dist/api-v2/admin/sales-channels/validators.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminPostSalesChannelsChannelProductsBatchReq = exports.AdminPostSalesChannelsSalesChannelReq = exports.AdminPostSalesChannelsReq = exports.AdminGetSalesChannelsParams = exports.AdminGetSalesChannelsSalesChannelParams = void 0;
var class_transformer_1 = require("class-transformer");
var class_validator_1 = require("class-validator");
var common_1 = require("../../../types/common");
var operator_map_1 = require("../../../types/validators/operator-map");
var AdminGetSalesChannelsSalesChannelParams = /** @class */ (function (_super) {
    __extends(AdminGetSalesChannelsSalesChannelParams, _super);
    function AdminGetSalesChannelsSalesChannelParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminGetSalesChannelsSalesChannelParams;
}(common_1.FindParams));
exports.AdminGetSalesChannelsSalesChannelParams = AdminGetSalesChannelsSalesChannelParams;
var AdminGetSalesChannelsParams = /** @class */ (function (_super) {
    __extends(AdminGetSalesChannelsParams, _super);
    function AdminGetSalesChannelsParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminGetSalesChannelsParams.prototype, "id", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminGetSalesChannelsParams.prototype, "name", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminGetSalesChannelsParams.prototype, "description", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)(),
        (0, class_transformer_1.Type)(function () { return operator_map_1.OperatorMapValidator; }),
        __metadata("design:type", Object)
    ], AdminGetSalesChannelsParams.prototype, "created_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)(),
        (0, class_transformer_1.Type)(function () { return operator_map_1.OperatorMapValidator; }),
        __metadata("design:type", Object)
    ], AdminGetSalesChannelsParams.prototype, "updated_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return AdminGetSalesChannelsParams; }),
        __metadata("design:type", Array)
    ], AdminGetSalesChannelsParams.prototype, "$and", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return AdminGetSalesChannelsParams; }),
        __metadata("design:type", Array)
    ], AdminGetSalesChannelsParams.prototype, "$or", void 0);
    return AdminGetSalesChannelsParams;
}((0, common_1.extendedFindParamsMixin)({
    limit: 20,
    offset: 0,
})));
exports.AdminGetSalesChannelsParams = AdminGetSalesChannelsParams;
var AdminPostSalesChannelsReq = /** @class */ (function () {
    function AdminPostSalesChannelsReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostSalesChannelsReq.prototype, "name", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostSalesChannelsReq.prototype, "description", void 0);
    __decorate([
        (0, class_validator_1.IsBoolean)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Boolean)
    ], AdminPostSalesChannelsReq.prototype, "is_disabled", void 0);
    __decorate([
        (0, class_validator_1.IsNotEmpty)(),
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminPostSalesChannelsReq.prototype, "metadata", void 0);
    return AdminPostSalesChannelsReq;
}());
exports.AdminPostSalesChannelsReq = AdminPostSalesChannelsReq;
var AdminPostSalesChannelsSalesChannelReq = /** @class */ (function () {
    function AdminPostSalesChannelsSalesChannelReq() {
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostSalesChannelsSalesChannelReq.prototype, "name", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostSalesChannelsSalesChannelReq.prototype, "description", void 0);
    __decorate([
        (0, class_validator_1.IsBoolean)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Boolean)
    ], AdminPostSalesChannelsSalesChannelReq.prototype, "is_disabled", void 0);
    __decorate([
        (0, class_validator_1.IsNotEmpty)(),
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminPostSalesChannelsSalesChannelReq.prototype, "metadata", void 0);
    return AdminPostSalesChannelsSalesChannelReq;
}());
exports.AdminPostSalesChannelsSalesChannelReq = AdminPostSalesChannelsSalesChannelReq;
var AdminPostSalesChannelsChannelProductsBatchReq = /** @class */ (function () {
    function AdminPostSalesChannelsChannelProductsBatchReq() {
    }
    __decorate([
        (0, class_validator_1.IsArray)(),
        __metadata("design:type", Array)
    ], AdminPostSalesChannelsChannelProductsBatchReq.prototype, "product_ids", void 0);
    return AdminPostSalesChannelsChannelProductsBatchReq;
}());
exports.AdminPostSalesChannelsChannelProductsBatchReq = AdminPostSalesChannelsChannelProductsBatchReq;
