// @medusajs/medusa/dist/api-v2/admin/collections/query-config.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listTransformQueryConfig = exports.retrieveTransformQueryConfig = exports.defaultAdminCollectionFields = exports.defaultAdminCollectionRelations = exports.allowedAdminCollectionRelations = void 0;
exports.allowedAdminCollectionRelations = ["products.profiles"];
// TODO: See how these should look when expanded
exports.defaultAdminCollectionRelations = ["products.profiles"];
exports.defaultAdminCollectionFields = [
    "id",
    "title",
    "handle",
    "created_at",
    "updated_at",
];
exports.retrieveTransformQueryConfig = {
    defaultFields: exports.defaultAdminCollectionFields,
    defaultRelations: exports.defaultAdminCollectionRelations,
    allowedRelations: exports.allowedAdminCollectionRelations,
    isList: false,
};
exports.listTransformQueryConfig = __assign(__assign({}, exports.retrieveTransformQueryConfig), { defaultLimit: 10, isList: true });
