// @medusajs/medusa/dist/api-v2/admin/collections/validators.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminPostCollectionsCollectionReq = exports.AdminPostCollectionsReq = exports.AdminGetCollectionsParams = exports.AdminGetCollectionsCollectionParams = void 0;
var class_transformer_1 = require("class-transformer");
var class_validator_1 = require("class-validator");
var common_1 = require("../../../types/common");
var operator_map_1 = require("../../../types/validators/operator-map");
// TODO: Ensure these match the DTOs in the types
var AdminGetCollectionsCollectionParams = /** @class */ (function (_super) {
    __extends(AdminGetCollectionsCollectionParams, _super);
    function AdminGetCollectionsCollectionParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminGetCollectionsCollectionParams;
}(common_1.FindParams));
exports.AdminGetCollectionsCollectionParams = AdminGetCollectionsCollectionParams;
/**
 * Parameters used to filter and configure the pagination of the retrieved regions.
 */
var AdminGetCollectionsParams = /** @class */ (function (_super) {
    __extends(AdminGetCollectionsParams, _super);
    function AdminGetCollectionsParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminGetCollectionsParams.prototype, "q", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", Object)
    ], AdminGetCollectionsParams.prototype, "title", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", Object)
    ], AdminGetCollectionsParams.prototype, "handle", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)(),
        (0, class_transformer_1.Type)(function () { return operator_map_1.OperatorMapValidator; }),
        __metadata("design:type", Object)
    ], AdminGetCollectionsParams.prototype, "created_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)(),
        (0, class_transformer_1.Type)(function () { return operator_map_1.OperatorMapValidator; }),
        __metadata("design:type", Object)
    ], AdminGetCollectionsParams.prototype, "updated_at", void 0);
    __decorate([
        (0, class_validator_1.ValidateNested)(),
        (0, class_validator_1.IsOptional)(),
        (0, class_transformer_1.Type)(function () { return operator_map_1.OperatorMapValidator; }),
        __metadata("design:type", Object)
    ], AdminGetCollectionsParams.prototype, "deleted_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return AdminGetCollectionsParams; }),
        __metadata("design:type", Array)
    ], AdminGetCollectionsParams.prototype, "$and", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return AdminGetCollectionsParams; }),
        __metadata("design:type", Array)
    ], AdminGetCollectionsParams.prototype, "$or", void 0);
    return AdminGetCollectionsParams;
}((0, common_1.extendedFindParamsMixin)({
    limit: 10,
    offset: 0,
})));
exports.AdminGetCollectionsParams = AdminGetCollectionsParams;
var AdminPostCollectionsReq = /** @class */ (function () {
    function AdminPostCollectionsReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsNotEmpty)(),
        __metadata("design:type", String)
    ], AdminPostCollectionsReq.prototype, "title", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostCollectionsReq.prototype, "handle", void 0);
    __decorate([
        (0, class_validator_1.IsObject)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminPostCollectionsReq.prototype, "metadata", void 0);
    return AdminPostCollectionsReq;
}());
exports.AdminPostCollectionsReq = AdminPostCollectionsReq;
var AdminPostCollectionsCollectionReq = /** @class */ (function () {
    function AdminPostCollectionsCollectionReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostCollectionsCollectionReq.prototype, "title", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostCollectionsCollectionReq.prototype, "handle", void 0);
    __decorate([
        (0, class_validator_1.IsObject)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminPostCollectionsCollectionReq.prototype, "metadata", void 0);
    return AdminPostCollectionsCollectionReq;
}());
exports.AdminPostCollectionsCollectionReq = AdminPostCollectionsCollectionReq;
