// @medusajs/core-flows/dist/product/workflows/create-products.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createProductsWorkflow = exports.createProductsWorkflowId = void 0;
const workflows_sdk_1 = require("@medusajs/workflows-sdk");
const steps_1 = require("../steps");
const pricing_1 = require("../../pricing");
exports.createProductsWorkflowId = "create-products";
exports.createProductsWorkflow = (0, workflows_sdk_1.createWorkflow)(exports.createProductsWorkflowId, (input) => {
    // Passing prices to the product module will fail, we want to keep them for after the product is created.
    const productWithoutPrices = (0, workflows_sdk_1.transform)({ input }, (data) => data.input.products.map((p) => ({
        ...p,
        variants: p.variants?.map((v) => ({
            ...v,
            prices: undefined,
        })),
    })));
    const createdProducts = (0, steps_1.createProductsStep)(productWithoutPrices);
    // Note: We rely on the same order of input and output when creating products here, make sure that assumption holds
    const variantsWithAssociatedPrices = (0, workflows_sdk_1.transform)({ input, createdProducts }, (data) => {
        return data.createdProducts
            .map((p, i) => {
            const inputProduct = data.input.products[i];
            return p.variants?.map((v, j) => ({
                id: v.id,
                prices: inputProduct?.variants?.[j]?.prices,
            }));
        })
            .flat()
            .filter((v) => !!v.prices?.length);
    });
    const createdPriceSets = (0, pricing_1.createPriceSetsStep)(variantsWithAssociatedPrices);
    const variantAndPriceSets = (0, workflows_sdk_1.transform)({ variantsWithAssociatedPrices, createdPriceSets }, (data) => data.variantsWithAssociatedPrices.map((variant, i) => ({
        variant: variant,
        price_set: data.createdPriceSets[i],
    })));
    const variantAndPriceSetLinks = (0, workflows_sdk_1.transform)({ variantAndPriceSets }, (data) => {
        return {
            links: data.variantAndPriceSets.map((entry) => ({
                variant_id: entry.variant.id,
                price_set_id: entry.price_set.id,
            })),
        };
    });
    (0, steps_1.createVariantPricingLinkStep)(variantAndPriceSetLinks);
    // TODO: Should we just refetch the products here?
    return (0, workflows_sdk_1.transform)({
        createdProducts,
        variantAndPriceSets,
    }, (data) => {
        return data.createdProducts.map((product) => ({
            ...product,
            variants: product.variants?.map((variant) => ({
                ...variant,
                price_set: data.variantAndPriceSets.find((v) => v.variant.id === variant.id)?.price_set,
            })),
        }));
    });
});
