// @medusajs/medusa/dist/api-v2/admin/api-keys/validators.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminDeleteApiKeysApiKeyReq = exports.AdminRevokeApiKeysApiKeyReq = exports.AdminPostApiKeysApiKeyReq = exports.AdminPostApiKeysReq = exports.AdminGetApiKeysParams = exports.AdminGetApiKeysApiKeyParams = void 0;
var class_transformer_1 = require("class-transformer");
var class_validator_1 = require("class-validator");
var common_1 = require("../../../types/common");
var utils_1 = require("@medusajs/utils");
var AdminGetApiKeysApiKeyParams = /** @class */ (function (_super) {
    __extends(AdminGetApiKeysApiKeyParams, _super);
    function AdminGetApiKeysApiKeyParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminGetApiKeysApiKeyParams;
}(common_1.FindParams));
exports.AdminGetApiKeysApiKeyParams = AdminGetApiKeysApiKeyParams;
/**
 * Parameters used to filter and configure the pagination of the retrieved api keys.
 */
var AdminGetApiKeysParams = /** @class */ (function (_super) {
    __extends(AdminGetApiKeysParams, _super);
    function AdminGetApiKeysParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, class_validator_1.IsString)({ each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminGetApiKeysParams.prototype, "id", void 0);
    __decorate([
        (0, class_validator_1.IsString)({ each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminGetApiKeysParams.prototype, "title", void 0);
    __decorate([
        (0, class_validator_1.IsString)({ each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminGetApiKeysParams.prototype, "token", void 0);
    __decorate([
        (0, class_validator_1.IsEnum)(utils_1.ApiKeyType, { each: true }),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminGetApiKeysParams.prototype, "type", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return AdminGetApiKeysParams; }),
        __metadata("design:type", Array)
    ], AdminGetApiKeysParams.prototype, "$and", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateNested)({ each: true }),
        (0, class_transformer_1.Type)(function () { return AdminGetApiKeysParams; }),
        __metadata("design:type", Array)
    ], AdminGetApiKeysParams.prototype, "$or", void 0);
    return AdminGetApiKeysParams;
}((0, common_1.extendedFindParamsMixin)({
    limit: 50,
    offset: 0,
})));
exports.AdminGetApiKeysParams = AdminGetApiKeysParams;
var AdminPostApiKeysReq = /** @class */ (function () {
    function AdminPostApiKeysReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostApiKeysReq.prototype, "title", void 0);
    __decorate([
        (0, class_validator_1.IsEnum)(utils_1.ApiKeyType, {}),
        __metadata("design:type", String)
    ], AdminPostApiKeysReq.prototype, "type", void 0);
    return AdminPostApiKeysReq;
}());
exports.AdminPostApiKeysReq = AdminPostApiKeysReq;
var AdminPostApiKeysApiKeyReq = /** @class */ (function () {
    function AdminPostApiKeysApiKeyReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostApiKeysApiKeyReq.prototype, "title", void 0);
    return AdminPostApiKeysApiKeyReq;
}());
exports.AdminPostApiKeysApiKeyReq = AdminPostApiKeysApiKeyReq;
var AdminRevokeApiKeysApiKeyReq = /** @class */ (function () {
    function AdminRevokeApiKeysApiKeyReq() {
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsNumber)(),
        __metadata("design:type", Number)
    ], AdminRevokeApiKeysApiKeyReq.prototype, "revoke_in", void 0);
    return AdminRevokeApiKeysApiKeyReq;
}());
exports.AdminRevokeApiKeysApiKeyReq = AdminRevokeApiKeysApiKeyReq;
var AdminDeleteApiKeysApiKeyReq = /** @class */ (function () {
    function AdminDeleteApiKeysApiKeyReq() {
    }
    return AdminDeleteApiKeysApiKeyReq;
}());
exports.AdminDeleteApiKeysApiKeyReq = AdminDeleteApiKeysApiKeyReq;
