// @medusajs/medusa/dist/api-v2/admin/pricing/query-config.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listTransformQueryConfig = exports.retrieveTransformQueryConfig = exports.defaultAdminPricingRuleTypeFields = exports.allowedAdminPricingRuleTypeRelations = exports.defaultAdminPricingRuleTypeRelations = void 0;
exports.defaultAdminPricingRuleTypeRelations = [];
exports.allowedAdminPricingRuleTypeRelations = [];
exports.defaultAdminPricingRuleTypeFields = [
    "id",
    "name",
    "rule_attribute",
    "default_priority",
];
exports.retrieveTransformQueryConfig = {
    defaultFields: exports.defaultAdminPricingRuleTypeFields,
    defaultRelations: exports.defaultAdminPricingRuleTypeRelations,
    allowedRelations: exports.allowedAdminPricingRuleTypeRelations,
    isList: false,
};
exports.listTransformQueryConfig = __assign(__assign({}, exports.retrieveTransformQueryConfig), { isList: true });
