// @medusajs/medusa/dist/api-v2/admin/pricing/validators.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminDeletePricingRuleTypesRuleTypeReq = exports.AdminPostPricingRuleTypesRuleTypeReq = exports.AdminPostPricingRuleTypesReq = exports.AdminGetPricingRuleTypesParams = exports.AdminGetPricingRuleTypesRuleTypeParams = void 0;
var class_validator_1 = require("class-validator");
var common_1 = require("../../../types/common");
var AdminGetPricingRuleTypesRuleTypeParams = /** @class */ (function (_super) {
    __extends(AdminGetPricingRuleTypesRuleTypeParams, _super);
    function AdminGetPricingRuleTypesRuleTypeParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminGetPricingRuleTypesRuleTypeParams;
}(common_1.FindParams));
exports.AdminGetPricingRuleTypesRuleTypeParams = AdminGetPricingRuleTypesRuleTypeParams;
var AdminGetPricingRuleTypesParams = /** @class */ (function (_super) {
    __extends(AdminGetPricingRuleTypesParams, _super);
    function AdminGetPricingRuleTypesParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Array)
    ], AdminGetPricingRuleTypesParams.prototype, "rule_attribute", void 0);
    return AdminGetPricingRuleTypesParams;
}((0, common_1.extendedFindParamsMixin)({
    limit: 100,
    offset: 0,
})));
exports.AdminGetPricingRuleTypesParams = AdminGetPricingRuleTypesParams;
var AdminPostPricingRuleTypesReq = /** @class */ (function () {
    function AdminPostPricingRuleTypesReq() {
    }
    __decorate([
        (0, class_validator_1.IsNotEmpty)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostPricingRuleTypesReq.prototype, "name", void 0);
    __decorate([
        (0, class_validator_1.IsNotEmpty)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostPricingRuleTypesReq.prototype, "rule_attribute", void 0);
    __decorate([
        (0, class_validator_1.IsNotEmpty)(),
        (0, class_validator_1.IsNumber)(),
        __metadata("design:type", Number)
    ], AdminPostPricingRuleTypesReq.prototype, "default_priority", void 0);
    return AdminPostPricingRuleTypesReq;
}());
exports.AdminPostPricingRuleTypesReq = AdminPostPricingRuleTypesReq;
var AdminPostPricingRuleTypesRuleTypeReq = /** @class */ (function () {
    function AdminPostPricingRuleTypesRuleTypeReq() {
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostPricingRuleTypesRuleTypeReq.prototype, "name", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostPricingRuleTypesRuleTypeReq.prototype, "rule_attribute", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsNumber)(),
        __metadata("design:type", Number)
    ], AdminPostPricingRuleTypesRuleTypeReq.prototype, "default_priority", void 0);
    return AdminPostPricingRuleTypesRuleTypeReq;
}());
exports.AdminPostPricingRuleTypesRuleTypeReq = AdminPostPricingRuleTypesRuleTypeReq;
var AdminDeletePricingRuleTypesRuleTypeReq = /** @class */ (function () {
    function AdminDeletePricingRuleTypesRuleTypeReq() {
    }
    return AdminDeletePricingRuleTypesRuleTypeReq;
}());
exports.AdminDeletePricingRuleTypesRuleTypeReq = AdminDeletePricingRuleTypesRuleTypeReq;
