// @medusajs/core-flows/dist/definition/cart/workflows/add-shipping-method-to-cart.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addShippingMethodToWorkflow = exports.addShippingMethodToCartWorkflowId = void 0;
const workflows_sdk_1 = require("@medusajs/workflows-sdk");
const use_remote_query_1 = require("../../../common/steps/use-remote-query");
const steps_1 = require("../steps");
const get_shipping_option_price_sets_1 = require("../steps/get-shipping-option-price-sets");
const refresh_cart_promotions_1 = require("../steps/refresh-cart-promotions");
const update_tax_lines_1 = require("../steps/update-tax-lines");
exports.addShippingMethodToCartWorkflowId = "add-shipping-method-to-cart";
exports.addShippingMethodToWorkflow = (0, workflows_sdk_1.createWorkflow)(exports.addShippingMethodToCartWorkflowId, (input) => {
    const optionIds = (0, workflows_sdk_1.transform)({ input }, (data) => {
        return (data.input.options ?? []).map((i) => i.id);
    });
    const priceSets = (0, get_shipping_option_price_sets_1.getShippingOptionPriceSetsStep)({
        optionIds: optionIds,
        context: { currency_code: input.currency_code },
    });
    const shippingOptions = (0, use_remote_query_1.useRemoteQueryStep)({
        entry_point: "shipping_option",
        fields: ["id", "name"],
        variables: {
            id: optionIds,
        },
    });
    const shippingMethodInput = (0, workflows_sdk_1.transform)({ priceSets, input, shippingOptions }, (data) => {
        const options = (data.input.options ?? []).map((option) => {
            const shippingOption = data.shippingOptions.find((so) => so.id === option.id);
            const price = data.priceSets[option.id].calculated_amount;
            return {
                shipping_option_id: shippingOption.id,
                amount: price,
                data: option.data ?? {},
                name: shippingOption.name,
                cart_id: data.input.cart_id,
            };
        });
        return options;
    });
    const shippingMethods = (0, steps_1.addShippingMethodToCartStep)({
        shipping_methods: shippingMethodInput,
    });
    (0, refresh_cart_promotions_1.refreshCartPromotionsStep)({ id: input.cart_id });
    (0, update_tax_lines_1.updateTaxLinesStep)({
        cart_or_cart_id: input.cart_id,
        shipping_methods: shippingMethods,
    });
});
