// @medusajs/medusa/dist/api-v2/admin/price-lists/validators.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminPriceListPricesUpdateReq = exports.AdminPostPriceListsPriceListPricesBatchRemoveReq = exports.AdminPostPriceListsPriceListPricesBatchAddReq = exports.AdminPostPriceListsPriceListReq = exports.AdminPriceListPricesCreateReq = exports.AdminPostPriceListsReq = exports.AdminGetPriceListsPriceListParams = exports.AdminGetPriceListsParams = void 0;
var types_1 = require("@medusajs/types");
var class_transformer_1 = require("class-transformer");
var class_validator_1 = require("class-validator");
var common_1 = require("../../../types/common");
var date_transform_1 = require("../../../utils/validators/date-transform");
var AdminGetPriceListsParams = /** @class */ (function (_super) {
    __extends(AdminGetPriceListsParams, _super);
    function AdminGetPriceListsParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminGetPriceListsParams;
}(common_1.FindParams));
exports.AdminGetPriceListsParams = AdminGetPriceListsParams;
var AdminGetPriceListsPriceListParams = /** @class */ (function (_super) {
    __extends(AdminGetPriceListsPriceListParams, _super);
    function AdminGetPriceListsPriceListParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminGetPriceListsPriceListParams;
}(common_1.FindParams));
exports.AdminGetPriceListsPriceListParams = AdminGetPriceListsPriceListParams;
var AdminPostPriceListsReq = /** @class */ (function () {
    function AdminPostPriceListsReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostPriceListsReq.prototype, "title", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostPriceListsReq.prototype, "description", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_transformer_1.Transform)(date_transform_1.transformOptionalDate),
        __metadata("design:type", String)
    ], AdminPostPriceListsReq.prototype, "starts_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_transformer_1.Transform)(date_transform_1.transformOptionalDate),
        __metadata("design:type", String)
    ], AdminPostPriceListsReq.prototype, "ends_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsEnum)(types_1.PriceListStatus),
        __metadata("design:type", String)
    ], AdminPostPriceListsReq.prototype, "status", void 0);
    __decorate([
        (0, class_validator_1.IsEnum)(types_1.PriceListType),
        __metadata("design:type", String)
    ], AdminPostPriceListsReq.prototype, "type", void 0);
    __decorate([
        (0, class_validator_1.IsArray)(),
        (0, class_transformer_1.Type)(function () { return AdminPriceListPricesCreateReq; }),
        (0, class_validator_1.ValidateNested)({ each: true }),
        __metadata("design:type", Array)
    ], AdminPostPriceListsReq.prototype, "prices", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsObject)(),
        __metadata("design:type", Object)
    ], AdminPostPriceListsReq.prototype, "rules", void 0);
    return AdminPostPriceListsReq;
}());
exports.AdminPostPriceListsReq = AdminPostPriceListsReq;
var AdminPriceListPricesCreateReq = /** @class */ (function () {
    function AdminPriceListPricesCreateReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPriceListPricesCreateReq.prototype, "currency_code", void 0);
    __decorate([
        (0, class_validator_1.IsInt)(),
        __metadata("design:type", Number)
    ], AdminPriceListPricesCreateReq.prototype, "amount", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPriceListPricesCreateReq.prototype, "variant_id", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsInt)(),
        __metadata("design:type", Number)
    ], AdminPriceListPricesCreateReq.prototype, "min_quantity", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsInt)(),
        __metadata("design:type", Number)
    ], AdminPriceListPricesCreateReq.prototype, "max_quantity", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsObject)(),
        __metadata("design:type", Object)
    ], AdminPriceListPricesCreateReq.prototype, "rules", void 0);
    return AdminPriceListPricesCreateReq;
}());
exports.AdminPriceListPricesCreateReq = AdminPriceListPricesCreateReq;
var AdminPostPriceListsPriceListReq = /** @class */ (function () {
    function AdminPostPriceListsPriceListReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostPriceListsPriceListReq.prototype, "title", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostPriceListsPriceListReq.prototype, "description", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_transformer_1.Transform)(date_transform_1.transformOptionalDate),
        __metadata("design:type", String)
    ], AdminPostPriceListsPriceListReq.prototype, "starts_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_transformer_1.Transform)(date_transform_1.transformOptionalDate),
        __metadata("design:type", String)
    ], AdminPostPriceListsPriceListReq.prototype, "ends_at", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsEnum)(types_1.PriceListStatus),
        __metadata("design:type", String)
    ], AdminPostPriceListsPriceListReq.prototype, "status", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsEnum)(types_1.PriceListType),
        __metadata("design:type", String)
    ], AdminPostPriceListsPriceListReq.prototype, "type", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsArray)(),
        __metadata("design:type", Array)
    ], AdminPostPriceListsPriceListReq.prototype, "prices", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsObject)(),
        __metadata("design:type", Object)
    ], AdminPostPriceListsPriceListReq.prototype, "rules", void 0);
    return AdminPostPriceListsPriceListReq;
}());
exports.AdminPostPriceListsPriceListReq = AdminPostPriceListsPriceListReq;
var AdminPostPriceListsPriceListPricesBatchAddReq = /** @class */ (function () {
    function AdminPostPriceListsPriceListPricesBatchAddReq() {
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsArray)(),
        __metadata("design:type", Array)
    ], AdminPostPriceListsPriceListPricesBatchAddReq.prototype, "prices", void 0);
    return AdminPostPriceListsPriceListPricesBatchAddReq;
}());
exports.AdminPostPriceListsPriceListPricesBatchAddReq = AdminPostPriceListsPriceListPricesBatchAddReq;
var AdminPostPriceListsPriceListPricesBatchRemoveReq = /** @class */ (function () {
    function AdminPostPriceListsPriceListPricesBatchRemoveReq() {
    }
    __decorate([
        (0, class_validator_1.IsArray)(),
        (0, class_validator_1.IsString)({ each: true }),
        __metadata("design:type", Array)
    ], AdminPostPriceListsPriceListPricesBatchRemoveReq.prototype, "ids", void 0);
    return AdminPostPriceListsPriceListPricesBatchRemoveReq;
}());
exports.AdminPostPriceListsPriceListPricesBatchRemoveReq = AdminPostPriceListsPriceListPricesBatchRemoveReq;
var AdminPriceListPricesUpdateReq = /** @class */ (function () {
    function AdminPriceListPricesUpdateReq() {
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPriceListPricesUpdateReq.prototype, "id", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateIf)(function (object) { return !object.id; }),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPriceListPricesUpdateReq.prototype, "currency_code", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateIf)(function (object) { return !object.id; }),
        (0, class_validator_1.IsInt)(),
        __metadata("design:type", Number)
    ], AdminPriceListPricesUpdateReq.prototype, "amount", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.ValidateIf)(function (object) { return !object.id; }),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPriceListPricesUpdateReq.prototype, "variant_id", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsInt)(),
        __metadata("design:type", Number)
    ], AdminPriceListPricesUpdateReq.prototype, "min_quantity", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsInt)(),
        __metadata("design:type", Number)
    ], AdminPriceListPricesUpdateReq.prototype, "max_quantity", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsObject)(),
        __metadata("design:type", Object)
    ], AdminPriceListPricesUpdateReq.prototype, "rules", void 0);
    return AdminPriceListPricesUpdateReq;
}());
exports.AdminPriceListPricesUpdateReq = AdminPriceListPricesUpdateReq;
